posthog/temporal/data_imports/pipelines/pipeline/pipeline.py

```diff
@@ -162,28 +163,34 @@ def run(self):
                 if isinstance(item, list):
                     if len(buffer) > 0:
                         buffer.extend(item)
-                        if asizeof.asizeof(buffer) >= self._chunk_size_bytes or len(buffer) >= self._chunk_size:
+                        buffer_size_bytes += _estimate_size(item)
+                        if buffer_size_bytes >= self._chunk_size_bytes or len(buffer) >= self._chunk_size:
                             self._logger.debug(f"Processing pipeline buffer (list). Length of buffer = {len(buffer)}")
 
                             py_table = table_from_py_list(buffer)
                             buffer = []
+                            buffer_size_bytes = 0
                         else:
                             continue
                     else:
-                        if asizeof.asizeof(item) >= self._chunk_size_bytes or len(item) >= self._chunk_size:
+                        buffer_size_bytes += _estimate_size(item)
+                        if buffer_size_bytes >= self._chunk_size_bytes or len(item) >= self._chunk_size:
                             self._logger.debug(f"Processing pipeline item (list). Length of item = {len(item)}")
                             py_table = table_from_py_list(item)
+                            buffer_size_bytes = 0
                         else:
                             buffer.extend(item)
                             continue
                 elif isinstance(item, dict):
                     buffer.append(item)
-                    if asizeof.asizeof(buffer) < self._chunk_size_bytes and len(buffer) < self._chunk_size:
+                    buffer_size_bytes += _estimate_size(item)
+                    if buffer_size_bytes < self._chunk_size_bytes and len(buffer) < self._chunk_size:
                         continue
 
                     self._logger.debug(f"Processing pipeline buffer (dict). Length of buffer = {len(buffer)}")
                     py_table = table_from_py_list(buffer)
                     buffer = []
+                    buffer_size_bytes = 0
                 elif isinstance(item, pa.Table):
                     py_table = item
```

Comment:
why buffer_size_bytes is not needed in this case?

Reply:
Because we're not buffering any items here - the item from the source is a pyarrow table - we only buffer list/dict items to then turn them into a pyarrow able

Reply:
thx